### DISCUSSION TITLE
Can kitty change my vim theme?

### DISCUSSION THREAD

**Discussion by ldelossa** [kovidgoyal/kitty#4226] [Q&A] [answered]
Hey there,

So I'm attempting to have a single shortcut on my machine change my system and vim theme. 

In vim, I have a key sequence bound to my theme change. 

Would it be possible for me to use the remote feature of kitty to send the key sequence "escape  dm" from an arbitrary shell (in my case the shell sway forks) targeting all kitty instances running "vim" ? 

I was going thru the tutorial and I ran into two blockers:

1. Calling "kitty @ send-text" seems to only work from the shell you call "kitty @ launch" from. 
2. I can't seem to find a way to send control characters such as "escape" with "send-text"

**Comment by page-down** [comment, marked as answer]
> 1. Calling "kitty @ send-text" seems to only work from the shell you call "kitty @ launch" from.

You can enable remote control via UNIX socket for kitty and then use the parameter `--to` when calling kitty @.

> 2. I can't seem to find a way to send control characters such as "escape" with "send-text"

As for sending to all VIM windows, you can first find out all the window ids by `-kitty @ ls` and send them exactly. `--match` also supports matching running processes.

As for escape, you can just send it directly.

```shell
kitty @ send-text -m id:1 "\x1b"
```

You can see the key escape sequence with the following command.

```shell
kitty +kitten show_key
```

**Reply by ldelossa** [reply]
thanks! I was attempting this with no luck:

```
shell zsh

allow_remote_control yes
listen_on unix:@mykitty
```

```
kitty @ --to unix:@mykitty send-text --match cmdline:nvim " lm"
Traceback (most recent call last):
  File "/usr/lib64/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib64/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/louis/git/c/kitty/kitty/launcher/../../__main__.py", line 153, in <module>
    main()
  File "/home/louis/git/c/kitty/kitty/launcher/../../__main__.py", line 149, in main
    func(sys.argv[1:])
  File "/home/louis/git/c/kitty/kitty/launcher/../../__main__.py", line 23, in remote_control
    rc_main(args)
  File "/home/louis/git/c/kitty/kitty/launcher/../../kitty/remote_control.py", line 243, in main
    response = do_io(global_opts.to, send, no_response, response_timeout)
  File "/home/louis/git/c/kitty/kitty/launcher/../../kitty/remote_control.py", line 152, in do_io
    with io:
  File "/home/louis/git/c/kitty/kitty/launcher/../../kitty/remote_control.py", line 99, in __enter__
    self.socket.connect(self.address)
ConnectionRefusedError: [Errno 111] Connection refused
```

I also tried with a regular old unix socket, but it created a socket like "kitty-123456" my expectation here is that I can write to a single socket to change all kitty instances with running "nvim", but is it actually the case that there is no main zygote process controlling all children kitty?

**Reply by page-down** [reply]
If you start many kitty processes, then each one will listen independently. For this case, you need to write scripts that check all processes and send them.

**Reply by page-down** [reply]
Have you tried the following parameter? Did it work for you? Open multiple windows in one process.
`--single-instance`
https://sw.kovidgoyal.net/kitty/invocation/#cmdoption-kitty-single-instance

Or use the remote control command to create a new window.
`kitty @ launch`
https://sw.kovidgoyal.net/kitty/remote-control/#kitty-launch

This may eliminate the need to run multiple processes.

**Reply by ldelossa** [reply]
I don't think "--single-instance" would work well, since I have many independent kitty instances running across multiple workspaces. The term "window" here is not an OS window right? Multiple windows means multiple pains in as single terminal in this case?

Either way, writing a script which writes to each unix sock is completely acceptable. I'll do this :)

**Reply by page-down** [reply]
Oh, I meant OS window. An instance can have more than one OS window.
This does not work if you want to have separate kitty processes (instances) in each workspace.
By all means, just use what works best for you.

**Reply by ldelossa** [reply]
Thanks for the help.

**Reply by ldelossa** [reply]
One more question, can I use the "--match" flag with the "--to" flag?

**Reply by page-down** [reply]
Yes, it works together, and you can also use the KITTY_LISTEN_ON environment variable instead of the --to parameter. Please refer to the documentation for details.

**Reply by ldelossa** [reply]
Okay it seems to be failing to find my nvim processes. I launch it with a simple "nvim" call. I however launch my kitty terminals from a sway key binding. If I remove the match flag it all works.
